user/tests/wiring/shared_interrupt/shared_interrupt.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2023 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "application.h"
+#include "unit-test/unit-test.h"
+
+#if HAL_PLATFORM_SHARED_INTERRUPT
+
+constexpr uint8_t handlers = 3;
+uint32_t record[handlers] = { 0 };
+uint8_t idx = 0;
+
+void handler(void* data) {
+    record[idx++] = (uint32_t)data;
+    idx %= handlers;
+}
+
+void handler1(void* data) {
+    handler(data);
+}
+
+test(01_Shared_Interrupt_Execute_Handlers_As_Expected) {
+    constexpr uint8_t trigPin = TX;
+    constexpr uint8_t intPin = WKP;
+    const InterruptMode mode = FALLING;
+
+    Serial.println("Please connect TX with WKP and then press any key to start the tests.");
+    while (Serial.available() <= 0);
+    while (Serial.available() > 0) {
+        (void)Serial.read();
+    }
+
+    hal_gpio_mode(trigPin, OUTPUT);
+    hal_gpio_write(trigPin, HIGH);
+
+    hal_gpio_mode(intPin, INPUT_PULLUP);
+    hal_interrupt_extra_configuration_t config = {};
+    config.flags = HAL_INTERRUPT_FLAG_APPEND_HANDLER;
+
+    config.chainPriority = 1;
+    hal_interrupt_attach(intPin, handler, (void*)0x12345678, mode, &config);
```

Comment:
Multiple handlers can be attached to a pin via `hal_interrupt_attach()` in system code, but users cannot currently append multiple handlers using `attachInterrupt()`.
Is this intentional? Do we only want to have this functionality available on the system side?

Reply:
We are not exposing this for general use in wiring as of right now, correct.